storagemarket/integration_test.go

```diff
@@ -277,65 +281,104 @@ func TestMakeDealNonBlocking(t *testing.T) {
 	// Provider should be blocking on waiting for funds to appear on chain
 	pd := providerDeals[0]
 	assert.Equal(t, result.ProposalCid, pd.ProposalCid)
-	shared_testutil.AssertDealState(t, storagemarket.StorageDealProviderFunding, pd.State)
+	require.Eventually(t, func() bool {
+		return pd.State == storagemarket.StorageDealProviderFunding
+	}, 1*time.Second, 100*time.Millisecond, "actual deal status is %s", storagemarket.DealStates[pd.State])
 }
 
+// FIXME Gets hung sometimes
 func TestRestartClient(t *testing.T) {
 	testCases := map[string]struct {
-		stopAtEvent storagemarket.ClientEvent
-		fh          func(h *harness)
+		stopAtEvent         storagemarket.ClientEvent
+		expectedClientState storagemarket.StorageDealStatus
+		clientDelay         testnodes.DelayFakeCommonNode
+		providerDelay       testnodes.DelayFakeCommonNode
+		fh                  func(h *harness)
 	}{
-		"ClientEventFundsEnsured": {
-			stopAtEvent: storagemarket.ClientEventFundsEnsured,
-		},
-		"ClientEventInitiateDataTransfer": {
-			stopAtEvent: storagemarket.ClientEventInitiateDataTransfer,
-		},
+
 		"ClientEventDataTransferInitiated": {
-			stopAtEvent: storagemarket.ClientEventDataTransferInitiated,
+			// This test can fail if client crashes without seeing a Provider DT complete
```

Comment:
@hannahhoward This is a very tricky one. If the client crashes before processing a `Complete Event` but the Provider has already completed it's FSM, the `Restart` will fail later as the channel will NOT exist in the Provider's FSM.

Reply:
Do we want to fix this ?